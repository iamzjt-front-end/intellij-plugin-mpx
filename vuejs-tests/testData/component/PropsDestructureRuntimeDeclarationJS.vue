<script setup>
const {
  prop2 = 1,
  prop3 = 1,
  prop4: aliased,
  ...rest
} = defineProps({
  prop1: { type: Number, required: true },
  prop2: { type: Number, required: true },
  prop3: Number,
  prop4: Number,
});

<weak_warning descr="Unresolved variable or type prop1">prop1</weak_warning>;
rest.prop1;
rest.prop2;
<weak_warning descr="Unresolved variable or type prop4">prop4</weak_warning>;
aliased;
</script>

<template>
  <<warning descr="Element PropsDestructureRuntimeDeclarationJS doesn't have required attribute prop1">PropsDestructureRuntimeDeclarationJS</warning> />
  <PropsDestructureRuntimeDeclarationJS :prop1="1" />
  <PropsDestructureRuntimeDeclarationJS :prop1="undefined" :prop2="undefined" :prop3="undefined" :prop4="undefined" />
  <PropsDestructureRuntimeDeclarationJS :prop1="1" :prop2="1" :prop3="1" :prop4="1" />
  <PropsDestructureRuntimeDeclarationJS :prop1="prop1" :prop2="prop2" :prop3="prop3" :prop4="prop4" />

  {{prop1}}
  {{prop2}}
  {{prop3}}
  {{prop4}}
  {{aliased}}
  {{rest.prop1}}
</template>
