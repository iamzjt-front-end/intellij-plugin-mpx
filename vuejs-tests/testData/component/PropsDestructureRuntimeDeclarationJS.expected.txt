{
 class: "VueSourceComponent",
 components: {
  PropsDestructureRuntimeDeclarationJS: <recursive value of class VueSourceComponent>,
 },
 props: [
  {
   class: "VueSourceInputProperty",
   name: "prop1",
   jsType: "number",
   required: true,
  },
  {
   class: "VueSourceInputProperty",
   name: "prop2",
   jsType: "number",
   required: false,
  },
  {
   class: "VueSourceInputProperty",
   name: "prop3",
   jsType: "number",
   required: false,
  },
  {
   class: "VueSourceInputProperty",
   name: "prop4",
   jsType: "number",
   required: false,
  },
 ],
}
